File: 백준/Silver/20291. 파일 정리/파일 정리.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "ex.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");

const n = Number(input[0]);

const file = new Map();

let answer = "";

for (let i = 1; i <= n; i++) {
  const [a, b] = input[i].trim().split(".");
  if (!file.has(b)) {
    file.set(b, 1);
  } else {
    file.set(b, file.get(b) + 1);
  }
}

const fileArray = [...file];
fileArray.sort();

for (const [key, value] of fileArray) {
  answer += `${key} ${value}\n`;
}

console.log(answer);

File: 백준/Silver/20291. 파일 정리/파일 정리.py
n = int(input())

answer = {}

for _ in range(n):
    a, b = input().split(".")
    if answer.get(b) == None:
        answer[b] = 1
    else:
        answer[b] += 1

sorted_answer = sorted(answer.items(), key=lambda item:item[0])

for key, value in sorted_answer:
    print(key, value)